### PR TITLE
Add back `Out-Printer` command

### LINKED ISSUE #10660: Bring back Out-Printer
# Summary of the new feature/enhancement

'As a user I want to be able to send output of a command to a printer.' 
For example `Get-Process | out-printer "OneNote"`
would save the current processes to a new page in OneNote 
or `Get-Package | lp`
Would send a list packages to the default printer 
# Proposed technical implementation details 
Windows PowerShell has an `Out-Printer` command with an alias of `lp `. 
but this was dropped in PowerShell core V6 
I have posted a version for V6 and later at https://www.powershellgallery.com/packages/6print with a repo at https://github.com/jhoneill/6Print 

This provides some additional controls over the output. 
I'd propose this as a first draft / worked example of how it could be done.

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

System System.Drawing is back, we can re-enable `Out-Printer` command.  Manually validated using `Out-Printer` to pdf printer.

## PR Context

Fix https://github.com/PowerShell/PowerShell/issues/10660

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/5003
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj [hunk 1]
```diff
@@ -24,8 +24,6 @@
     <Compile Remove="commands\utility\FormatAndOutput\OutGridView\OutWindowProxy.cs" />
```
Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: @SteveL-MSFT, your last commit had 3 failures in `PowerShell-CI-macos`
Test-Connection.Ping.Force IPv4 with implicit PingOptions
```powershell
Cannot index into a null array.
at <ScriptBlock>, /Users/runner/runners/2.159.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 91
91:             $result.Replies[0].Address              | Should -BeExactly $realAddress

```
Test-Connection.Ping.Force IPv4 with explicit PingOptions
```powershell
Cannot index into a null array.
at <ScriptBlock>, /Users/runner/runners/2.159.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 106
106:             $result1.Replies[0].Address              | Should -BeExactly $realAddress

```
Test-Connection.TraceRoute.TraceRoute works
```powershell
Cannot index into a null array.
at <ScriptBlock>, /Users/runner/runners/2.159.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 238
238:             $pingReplies = $replies[-1].PingReplies

```

Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-windows`
Verify approved aliases list.All approved aliases have the correct 'ReadOnly' option
```powershell
Expected $null or empty, but got @(@{InputObject=lp -> Out-Printer; SideIndicator==>}).
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Basic\DefaultCommands.Tests.ps1: line 577
577:         $result | Should -BeNullOrEmpty
```

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 2]
```diff
@@ -4590,6 +4590,7 @@ internal static SessionStateAliasEntry[] BuiltInAliases
                     // These were transferred from the "transferred from the profile" section
                     new SessionStateAliasEntry("cat", "Get-Content"),
                     new SessionStateAliasEntry("cp", "Copy-Item", string.Empty, AllScope),
+                    new SessionStateAliasEntry("lp", "Out-Printer"),
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: Do we really want the alias? It conflicts with Unix lp.

Comment (iSazonov) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: The alias conflicts with Unix command. Do we really want the alias?

Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/InitialSessionState.cs]: Let's have @PowerShell/powershell-committee make a call here

Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/InitialSessionState.cs]: @PowerShell/powershell-committee reviewed this and agrees to remove the alias

**Comment by sdwheeler** [comment]
@SteveL-MSFT The new FWLink for Out-Printer is `https://go.microsoft.com/fwlink/?LinkID=2109553`

File: src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj (hunk 1 above)
Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-windows`
Online help tests for PowerShell Cmdlets.Validate 'get-help Out-Printer -Online'
```powershell
Expected strings to be the same, but they were different.
Expected length: 46
Actual length:   47
Strings differ at index 40.
Expected: '...nkID=11336...'
But was:  '...nkID=21095...'
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Help\HelpSystem.OnlineHelp.Tests.ps1: line 42
42:                 $actualURI | Should -Be $cmdlet.HelpURI
```

Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-static-analysis`
Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/learning-powershell/README.md.https://channel9.msdn.com/Series/GetStartedPowerShell3 should work
```powershell
retry of URL failed with error: Connection timed out
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 117
117:                                     throw "retry of URL failed with error: $($_.Exception.Message)"

```

**Comment by SteveL-MSFT** [comment]
@PoshChan please retry static

**Comment by PoshChan** [comment]
@SteveL-MSFT, successfully started retry of `PowerShell-CI-static-analysis`

File: test/powershell/engine/Basic/DefaultCommands.Tests.ps1 [hunk 3]
```diff
@@ -107,7 +107,6 @@ Describe "Verify approved aliases list" -Tags "CI" {
 "Alias",        "iwmi",                             "Invoke-WMIMethod",                 $($FullCLR                               ),     "ReadOnly",             "",                     ""
 "Alias",        "iwr",                              "Invoke-WebRequest",                $($FullCLR -or $CoreWindows -or $CoreUnix),     "ReadOnly",             "",                     ""
 "Alias",        "kill",                             "Stop-Process",                     $($FullCLR -or $CoreWindows              ),     "",                     "",                     ""
-"Alias",        "lp",                               "Out-Printer",                      $($FullCLR                               ),     "ReadOnly",             "",                     ""
```
Comment (iSazonov) [inline test/powershell/engine/Basic/DefaultCommands.Tests.ps1]: I'd keep this for FullCLR - this allows us to see version differences.

Comment (SteveL-MSFT) [reply test/powershell/engine/Basic/DefaultCommands.Tests.ps1]: Makes sense, will add back original

File: test/powershell/engine/Basic/DefaultCommands.Tests.ps1 [hunk 4]
```diff
@@ -523,8 +523,8 @@ Describe "Verify approved aliases list" -Tags "CI" {
             if ($isPreview) {
                 $emptyConfigPath = Join-Path -Path $TestDrive -ChildPath "test.config.json"
                 Set-Content -Path $emptyConfigPath -Value "" -Force -ErrorAction Stop
-                $currentAliasList = pwsh -out XML -SettingsFile $emptyConfigPath -Command $getAliases -args ($moduleList | ConvertTo-Json)
-                $currentCmdletList = pwsh -out XML -SettingsFile $emptyConfigPath -Command $getCommands -args ($moduleList | ConvertTo-Json)
+                $currentAliasList = pwsh -noprofile -out XML -SettingsFile $emptyConfigPath -Command $getAliases -args ($moduleList | ConvertTo-Json)
```
Comment (iSazonov) [inline test/powershell/engine/Basic/DefaultCommands.Tests.ps1]:
```suggestion
                $currentAliasList = pwsh -NoProfile -OutputFormat XML -SettingsFile $emptyConfigPath -Command $getAliases -args ($moduleList | ConvertTo-Json)
```

File: test/powershell/engine/Basic/DefaultCommands.Tests.ps1 [hunk 5]
```diff
@@ -523,8 +523,8 @@ Describe "Verify approved aliases list" -Tags "CI" {
...
+                $currentCmdletList = pwsh -noprofile -out XML -SettingsFile $emptyConfigPath -Command $getCommands -args ($moduleList | ConvertTo-Json)
```
Comment (iSazonov) [inline test/powershell/engine/Basic/DefaultCommands.Tests.ps1]:
```suggestion
                $currentCmdletList = pwsh -NoProfile -OutputFormat XML -SettingsFile $emptyConfigPath -Command $getCommands -args ($moduleList | ConvertTo-Json)
```

**Review by iSazonov — APPROVED** [review]
LGTM with two minor comments.

**Review by adityapatwardhan — APPROVED** [review]
Please respond to comments from @iSazonov. Otherwise looks good.

**Comment by adityapatwardhan** [comment]
@SteveL-MSFT Please resolve merge conflicts

**Comment by SteveL-MSFT** [comment]
I'll wait for GraphicalHost to be merged first as there will be another merge conflict

**Comment by adityapatwardhan** [comment]
@SteveL-MSFT Graphical host is merged. Please resolve merge conflict.

**Comment by adityapatwardhan** [comment]
@PoshChan please remind me in 1 hour.

**Comment by PoshChan** [comment]
@adityapatwardhan, this is the reminder you requested 1 hour. ago

**Pull request merged**